test/VaultWithLiquity.spec.ts

```diff
@@ -154,53 +154,99 @@ describe('VaultWithLiquity', () => {
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [
+          claimParams.percent(50).to(alice.address).build(),
+          claimParams.percent(50).to(bob.address).build(),
+        ],
+      });
+
+      await vault.connect(alice).deposit(params);
+
+      await vault.updateInvested();
+
+      // add 100 ETH to the strategy
+      setBalance(strategy.address, parseUnits('100'));
+
+      // swap 10 ETH for LUSD and reinvest
+      const swapData = ethers.utils.defaultAbiCoder.encode(
+        ['address', 'address', 'uint256'],
+        [
+          constants.AddressZero, // from
+          underlying.address, // to
+          parseUnits('10'),
+        ],
+      );
+
+      await strategy.reinvest(
+        mock0x.address,
+        0,
+        [],
+        parseUnits('10'),
+        swapData,
+        parseUnits('10'),
+      );
 
-    const params = depositParams.build({
-      amount: parseUnits('100'),
-      inputToken: underlying.address,
-      claims: [
-        claimParams.percent(50).to(alice.address).build(),
-        claimParams.percent(50).to(bob.address).build(),
-      ],
+      await vault.connect(alice).claimYield(alice.address);
+
+      expect(await underlyingBalanceOf(alice)).to.eq(parseUnits('0'));
+      expect(await getETHBalance(strategy.address)).to.eq(parseUnits('40'));
+      // we have to ignore the gas cost for alice here
+      expect(
+        (await getETHBalance(alice.address)).sub(alicesInitialEthBalace),
+      ).to.gte(parseUnits('4999', 16));
+      expect(await strategy.investedAssets()).to.eq(parseUnits('150'));
     });
 
-    await vault.connect(alice).deposit(params);
+    it('transfers yield in LUSD from the strategy to the user when ETH balance < yield amount', async () => {
+      await addUnderlyingBalance(alice, '100');
+      const alicesInitialEthBalace = await getETHBalance(alice.address);
 
-    await vault.updateInvested();
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
 
-    // add 100 ETH to the strategy
-    setBalance(strategy.address, parseUnits('100'));
+      await vault.connect(alice).deposit(params);
 
-    // swap 10 ETH for LUSD and reinvest
-    const swapData = ethers.utils.defaultAbiCoder.encode(
-      ['address', 'address', 'uint256'],
-      [
-        constants.AddressZero, // from
-        underlying.address, // to
-        parseUnits('10'),
-      ],
-    );
+      await vault.updateInvested();
 
-    await strategy.reinvest(
-      mock0x.address,
-      0,
-      [],
-      parseUnits('10'),
-      swapData,
-      parseUnits('10'),
-    );
+      // add 100 ETH to the strategy
+      setBalance(strategy.address, parseUnits('100'));
+
+      // swap 10 ETH for LUSD and reinvest
+      const swapData = ethers.utils.defaultAbiCoder.encode(
```

Comment:
instead of a comment, maybe we can move this to a helper function at the bottom?

Reply:
added a helper [function](https://github.com/lindy-labs/sc_solidity-contracts/blob/0821703f84d66ad642a75ea14bdd6937541db5fe/test/strategy/liquity/LiquityStrategy.spec.ts#L934) for creating swap data